test/integration/protocol_integration_test.cc

```diff
@@ -123,13 +123,17 @@ TEST_P(DownstreamProtocolIntegrationTest, RouterRedirect) {
 }
 
 TEST_P(ProtocolIntegrationTest, UnknownResponsecode) {
+  config_helper_.addRuntimeOverride(
+      "envoy.reloadable_features.dont_add_content_length_for_bodiless_requests", "true");
   initialize();
 
   codec_client_ = makeHttpConnection(lookupPort("http"));
 
   Http::TestResponseHeaderMapImpl response_headers{{":status", "600"}};
   auto response = sendRequestAndWaitForResponse(default_request_headers_, 0, response_headers, 0);
 
+  // Regression test https://github.com/envoyproxy/envoy/issues/14890 - no content-length added.
+  EXPECT_EQ(upstream_request_->headers().ContentLength(), nullptr);
```

Comment:
Should we have a dedicated test for this? Seems like we're just reusing an existing test

Reply:
I'm a fan of reusing existing tests where possible, just because it avoids extra CI load.  I can copy-paste-dup if you'd prefer!

Reply:
Seems okay as is!

Reply:
Possible preference for adding this check to a test case focuses on the happy path while proxying a GET request to a backend, but this is fine.